File: PA1_Server/src/edu/ucdenver/tournament/LineUp.java
package edu.ucdenver.tournament;

import java.io.Serializable;
import java.util.ArrayList;

/**
 * <p>This defines the Lineup class
 * which holds all the methods and attributes
 * of the Lineups. This class
 * is responsible for adding players to a lineup
 * and retreiving necessary information.
 * </p>
 * @author Karim Kabbara
 * @version 1.0
 * @since 2022-11-11
 */
public class LineUp implements Serializable {

    /**
     * <p>
     *     Team object for the lineup
     * </p>
     */
    private final Team team;

    /**
     * <p>
     *     List of players in the lineup
     * </p>
     */
    private final ArrayList<Player> lineUpPlayers;

    /**
     * <p>
     * This is the constructor
     * for the LineUp class. It sets
     * the team and initializes the player
     * list
     * </p>
     * @param team Team Object
     */
    public LineUp(Team team) {
        this.team = team;
        this.lineUpPlayers = new ArrayList<>();
    }

    /**
     * <p>
     * Gets the team for the lineup
     * </p>
     * @return Team object
     * */
    public Team getTeam() {
        return this.team;
    }

    /**
     * <p>
     * List of players in the lineup
     * </p>
     * @return List of players
     * */
    public ArrayList<Player> getPlayers() {
        return this.lineUpPlayers;
    }

    /**
     * <p>
     * Adds a player to the lineup.
     * Ensures lineup size is less than 11
     * before adding
     * </p>
     * @param player Player Object
     * */
    public void addPlayer(Player player) {
        try{
            if (this.lineUpPlayers.size() < 11){
                this.lineUpPlayers.add(player);
            }
            else throw new RuntimeException();
        }
        catch (RuntimeException rte){
            System.err.println("Lineup is Full!");
        }
    }

    /**
     * <p>
     * Gets all player's names in the lineup
     * </p>
     * @return String of all players' names
     * */
    @Override
    public String toString() {
        String lineUp = "";
        for (Player p: this.lineUpPlayers){
            lineUp += (p.getName() + "\n");
        }
        return lineUp;
    }
}

File: Server (Python)/src/LineUp.py
from Team import Team
from Player import Player


class LineUp:
    """
    This defines the Lineup class
    which holds all the methods and attributes
    of the Lineups. This class
    is responsible for adding players to a lineup
    and retrieving necessary information.
    """
    _team: Team # team object
    _line_up_players: list  # players in lineup

    def __init__(self, team: Team):
        """
        This is the constructor
        for the LineUp class. It sets
        the team and initializes the player list
        :param team: team object
        """
        self._team = team
        self._line_up_players = []

    @property
    def get_team(self) -> Team:
        """
        Gets the team for the lineup
        :return: Team object (Team)
        """
        return self._team

    @property
    def get_players(self) -> []:
        """
        Gets players in the lineup
        :return: Players (List)
        """
        return self._line_up_players

    def add_player(self, player: Player):
        """
        Adds a player to a lineup
        :param player: Player object
        """
        if len(self._line_up_players) < 11:  # if list has 10 or fewer players
            self._line_up_players.append(player)  # add player to line_up
        else:
            pass

    def __str__(self) -> str:
        """
        String representation of lineup
        :return: lineup as String (String)
        """
        line_up = ""
        for players in self._line_up_players:  # for each player in list
            line_up += (players.__name__ + "\n")  # print player name, then enter
        return line_up
